src/django_components/dependencies.py

```diff
@@ -672,39 +709,38 @@ def _gen_exec_script(
     # Components.manager.markScriptLoaded("js", "/abc/def3.js"),
     # ```
     #
-    # NOTE: It would be better to pass only the URL itself for `loadScript`, instead of a whole tag.
-    # But because we allow users to specify the Media class, and thus users can
-    # configure how the `<link>` or `<script>` tags are rendered, we need pass the whole tag.
+    # NOTE: It would be better to pass only the URL itself for `loadJs/loadCss`, instead of a whole tag.
+    #    But because we allow users to specify the Media class, and thus users can
+    #    configure how the `<link>` or `<script>` tags are rendered, we need pass the whole tag.
     #
-    # NOTE 2: We must NOT await for the Promises, otherwise we create a deadlock
-    # where the script loaded with `loadScript` (loadee) is inserted AFTER the script with `loadScript` (loader).
-    # But the loader will NOT finish, because it's waiting for loadee, which cannot start before loader ends.
+    # NOTE 2: We must NOT await for the Promises, otherwise we create a deadlock where
+    #    the script loaded with `loadJs` (loadee) would be inserted only AFTER the script that
+    #    called `loadJs()` (loader) has finished.
+    #    But the loader will NOT finish, because it's waiting for loadee, which cannot start before loader ends.
     escaped_to_load_js_tags = [_escape_js(tag, eval=False) for tag in to_load_js_tags]
     escaped_to_load_css_tags = [_escape_js(tag, eval=False) for tag in to_load_css_tags]
 
     # Make JS array whose items are interpreted as JS statements (e.g. functions)
     def js_arr(lst: List) -> str:
         return "[" + ", ".join(lst) + "]"
 
-    exec_script: types.js = f"""(() => {{
-        const loadedJsScripts = {json.dumps(loaded_js_urls)};
-        const loadedCssScripts = {json.dumps(loaded_css_urls)};
-        const toLoadJsScripts = {js_arr(escaped_to_load_js_tags)};
-        const toLoadCssScripts = {js_arr(escaped_to_load_css_tags)};
-
-        loadedJsScripts.forEach((s) => Components.manager.markScriptLoaded("js", s));
-        loadedCssScripts.forEach((s) => Components.manager.markScriptLoaded("css", s));
-
-        Promise.all(
-            toLoadJsScripts.map((s) => Components.manager.loadScript("js", s))
-        ).catch(console.error);
-        Promise.all(
-            toLoadCssScripts.map((s) => Components.manager.loadScript("css", s))
-        ).catch(console.error);
-    }})();
+    # NOTE: Wrap the body in self-executing function to avoid polluting the global scope.
+    exec_script: types.js = f"""
+        (() => {{
+            Components.manager._loadComponentScripts({{
+                loadedCssUrls: {json.dumps(loaded_css_urls)},
+                loadedJsUrls: {json.dumps(loaded_js_urls)},
+                toLoadCssTags: {js_arr(escaped_to_load_css_tags)},
+                toLoadJsTags: {js_arr(escaped_to_load_js_tags)},
+            }});
+            document.currentScript.remove();
```

Comment:
This line makes it so that after the script is executed the parent `<script>` is removed from the HTML.

This is relevant mostly for the HTML fragment use case, where people may be surprised by finding extra `<script>` tags, like seen here https://github.com/EmilStenstrom/django-components/issues/789